proposals/NNNN-resolve-distributed-actor-protocols.md

```diff
@@ -0,0 +1,474 @@
+
+## Detailed design
+
+The `@DistributedProtocol` macro generates a concrete `distributed actor` declaration as well as an extension which implements the protocol's method requirements with "stub" implementations.
+
+>  **NOTE:** The exact details of the macro synthesized code are not guaranteed to remain the same, and may change without notice.
+
+This proposal also introduces an empty `DistributedActorStub` protocol:
+
+```swift
+public protocol DistributedActorStub where Self: DistributedActor {}
+```
+
+The `@DistributedProtocol` macro synthesizes a concrete distributed actor which accepts a generic `ActorSystem`. The generated actor declaration matches access level with the original declaration, and implements the protocol as well as the `DistributedStub` protocol:
+
+```swift
+protocol Greeter: where ActorSystem: DistributedActorSystem<any Codable> {
+  distributed func greet(name: String) -> String
+}
+
+// "stub" type
+distributed actor $Greeter<ActorSystem>: Greeter, DistributedStubActor
+    where DistributedActorSystem<any Codable> {
+  private init() {} // cannot initialize, can only resolve(id:using:)
+}
+
+extension Greeter where Self: DistributedActorStub {
+  // ... stub implementations for protocol requirements ...
+}
+```
+
+Default implementations for all the protocol's requirements (including non-distributed requirements) are provided by extensions utilizing the `DistributedActorStub` protocol.
+
+It is possible for a protocol type to inherit other protocols, in that case the parent protocol must either have default implementations for all its requirements, or it must also apply the `@DistributedProtocol` protocol which generates such default implementations.
+
+The default method "stub" implementations provided by the `@DistributedProtocol` simply fatal error if they were to ever be invoked. In practice, invoking those methods is not possible, because resolving a stub will always return a remote reference, and therefore calls on these methods are redirected to `DistributedActorSystem`'s `remoteCall` rather than invoking the "local" methods.  
+
+It is recommended to use `some/any Greeter` rather than `$Greeter` when passing around resolved instances of a distributed actor protocol. This way none of your code is tied to the fact of using a specific type of proxy, but rather, can accept any greeter, be it local or resolved through a proxy reference. This can come in handy when refactoring a codebase, and merging modules in such way where the greeter may actually be a local instance in some situations.
+
+### Interaction with the `DefaultDistributedActorSystem`
+
+Since the introduction of distributed actors, it is possible to declare a module wide `DefaultDistributedActorSystem` type alias, like this:
```

Comment:
```suggestion
Since the introduction of distributed actors, it is possible to declare a module-wide `DefaultDistributedActorSystem` type alias, like this:
```